docker/gremlin-server/gremlin-server-integration.yaml

```diff
@@ -18,13 +18,15 @@
 host: 0.0.0.0
 port: 45940
 evaluationTimeout: 30000
+channelizer: org.apache.tinkerpop.gremlin.server.channel.WsAndHttpChannelizer
 graphs: {
-  graph: conf/tinkergraph-empty.properties,
-  classic: conf/tinkergraph-empty.properties,
-  modern: conf/tinkergraph-empty.properties,
-  crew: conf/tinkergraph-empty.properties,
-  grateful: conf/tinkergraph-empty.properties,
-  sink: conf/tinkergraph-empty.properties
+  graph: scripts/tinkergraph-empty.properties,
```

Comment:
It is because there is a difference between the `tinkergraph-empty.properties` file in the `conf/` and the `script/` folders. The `conf/` folder holds the default Gremlin Server set up, and the `script/` folder holds the test server set up (it is under the Gremlin Server `test/` folder). 

Javascript tests needed the file from `scripts` folder where the `vertexIdManager` is set to INTEGER, and we wanted to reuse this server yaml file, so it was changed here. 

Would you foresee problems with changing this file? I'm not sure if there was a particular reason for default to be different, perhaps that's also something to look into next.

Reply:
to avoid the confusion here, perhaps `scripts/tinkergraph-empty.properties` should be renamed and kept in `conf/`. maybe call it `conf/tinkergraph-service.properties`. a pity we need `INTEGER` ID manager in there for testing but i guess it's ok. 

historically, i think this file was kept separately to avoid it being added to the zip file binary packaging for releases. naming it `tinkergraph-service.properties` might allow it to be ok to be included as a test file as it would allow the `call()` step to be used (not sure I remember how that is setup now). only trouble is that `INTEGER` is not a good default for distribution, but maybe that's ok. I forget why `INTEGER` was needed but i'm almost sure that if you change that, tests will start to fail.

in any case, i think we should get all the graph configs pointed at `conf/` and not have this weird exception.

Reply:
Sounds good, I will take a look and make the changes needed.

Reply:
Hi @spmallette, I dug a bit into to see why `INTEGER` was needed for the javascript tests, and it looks like it's due to javascript deserializing `LONG` type ids into its native `Number` type, and storing the vertex ID as a `Number`:
```
getting vertex object from server
{
  '@type': 'g:Vertex',
  '@value': { id: { '@type': 'g:Int64', '@value': 6 }, label: 'person' }
}
deserialized vertex id is of type
number
```
So when the cucumber test queries the graph by the vertex, javascript serializes the `Number` vertex ID as an `INTEGER`, and since TinkerGraph requires the exact type of vertex IDs to match(i.e. a `LONG`), we get empty result and failed tests:
```
serialize vertex id is of type
number
serializing this
1
serializing this
{ '@type': 'g:Vertex', '@value': { id: 1, label: 'person' } }
serializing this
{ '@type': 'g:Bytecode', '@value': { step: [ [Array], [Array] ] } }
listing result for
g.V(v1).bothE()
[]
F-
```
So this forces javascript tests to use a server configuration of `gremlin.tinkergraph.vertexIdManager=INTEGER`  or `gremlin.tinkergraph.vertexIdManager=ANY` in order to pass the cucumber tests. The same failure will occur with GraphBinary as the logic runs the same. 

This looks specific to javascript due to the Number typing. I’ve tested this with the go driver to confirm, which passes cucumber tests with either `LONG` or `INTEGER` vertexIdManager, as it deserializes the vertex IDs into matching types.

Given we have made a specific `.properties` file for these tests, I'd assume this has been a known issue. So I'm wondering if we want to leave this issue alone for now, and just created a `conf/tinkergraph-service.properties` with `gremlin.tinkergraph.vertexIdManager=INTEGER` for javascript? Or do we want to resolve this typing issue in javascript first, so that we can use the default `.properties` file with `gremlin.tinkergraph.vertexIdManager=LONG`, and save from having to create the extra `.properties` file? 

I'm assuming we want to keep the `gremlin.tinkergraph.vertexIdManager=LONG` in the default `.properties` file for packaging as per this [commit](https://github.com/apache/tinkerpop/commit/8d10b81b8635d7ee256021256f6e9384da6c8744)

Thank you!

Reply:
Also just to add, I've only looked into this in the context of the failed tests, so there could be other concerns/implications with the `Number` types in javascript that I don't know. It may be the best to just add a `conf/tinkergraph-service.properties` for now.

Reply:
>  So I'm wondering if we want to leave this issue alone for now, and just created a conf/tinkergraph-service.properties with gremlin.tinkergraph.vertexIdManager=INTEGER for javascript?

yes, just keep the config as it is but rename the file.

separately, please create an issue in JIRA with your findings about Javascript. i assume the fix would be to properly generate Javascript to use `toLong()` somehow:

https://github.com/apache/tinkerpop/blob/624cfe87e275fe0d6dba710288471cc5b52c93f8/gremlin-javascript/src/main/javascript/gremlin-javascript/lib/utils.js#L28-L30

Reply:
Added the file into `conf` folder in this [34de230](https://github.com/apache/tinkerpop/pull/1788/commits/34de23022411f9109447632a83a6f5713397a02d). 

Issue created for javascript: https://issues.apache.org/jira/browse/TINKERPOP-2791